### PR TITLE
Backup Restore: Add a new parameter --ddl-batch-size for feature batch ddl

### PR DESCRIPTION
**Pull request by fengou1**
- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

A new parameter add for speed up restore

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by IANTHEREAL** [comment]
please not export this parameters to user

**Comment by TomShawn** [comment]
@fengou1 @IANTHEREAL @3pointer Shall we close this PR?

File: br/backup-and-restore-faq.md [hunk 1]
```diff
@@ -177,4 +177,11 @@ BR v4.0.9 备份统计信息使 BR 消耗过多内存，为保证备份过程正
...
-+ 多个 BR 并行恢复的场景没有经过测试，无法保证成功。
\ No newline at end of file
++ 多个 BR 并行恢复的场景没有经过测试，无法保证成功。
+
+## BR 带参数 `--ddl-batch-size` 恢复存档，立即遇到如下错误
+`entry too large, the max entry size is 6291456, the size of data is 7690800`
++ BR 带参数 `--ddl-batch-size` 恢复数据时，会调用内部接口 `BatchCreateTableWithInfo` 使用内部 ddl job `ActionCreateTables`, 每次按客户指定的 `--ddl-batch-size` 大小建表.
++ 因为 TiDB 在执行 DDL Job 时会把 Job 队列写到 TiKV 上，当前 TiDB 一次发送 Job message 最大值为 6MB (不推荐客户修改此值)，因此一次 batch size 的所有表的schema 总和不得超过 6MB。
++ 详细的限制请参考 TiKV 配置文件描述 raft-entry-max-size, 以及 TiDB 配置文件描述 txn-entry-size-limit 
++ 推荐客户使用 `--ddl-batch-size=128`
```
Comment (en-jin19) [inline br/backup-and-restore-faq.md:187]:
```suggestion
## BR 恢复存档时报错 `entry too large, the max entry size is 6291456, the size of data is 7690800`，该如何处理？

通过 【XXX】 方法，把 `--ddl-batch-size` 的值设置为 `128`。

你可以通过 【XXX】 方式确认 BR 是否已带有 `--ddl-batch-size` 参数。

当你已经设置了 `--ddl-batch-size` 参数值后，在使用 BR 恢复数据时，BR 会先调用内部接口 `BatchCreateTableWithInfo`，然后使用内部 DDL job `ActionCreateTables`，最后根据已设置的 `--ddl-batch-size` 的值来在 TiDB 建表。

当 TiDB 执行这个 DDL job 时，TiDB 会把 job 队列写到 TiKV 上。由于 TiDB 能够一次性发送的 job message 默认的最大值为 6 MB（**不建议**修改此值），因此，单次发送的所有表的 schema 总和不可以超过 6 MB，否则，会报 `entry too large, the max entry size is 6291456, the size of data is 7690800` 错误。

如需了解有关单个写入请求的数据量的具体限制，请参阅 TiDB 配置项 [txn-entry-size-limit](/tidb-configuration-file.md#txn-entry-size-limit-span-classversion-mark从-v50-版本开始引入span) 和 TiKV 配置项 [raft-entry-max-size](/tikv-configuration-file.md#raft-entry-max-size)。

```

Comment (en-jin19) [reply br/backup-and-restore-faq.md:187]: 该部分放到 165 行部分如何呢？因为 166 行的内容也是关于BR 恢复存档的。

**Comment by en-jin19** [comment]
This PR was closed and the corresponding feature document was updated in https://github.com/pingcap/docs-cn/pull/8659. Since this PR is no longer to be reviewed and updated, the translation/doing label is removed.

**Pull request closed without merging**